utils/dumpread/src/mapfile.cpp

```diff
@@ -0,0 +1,230 @@
+// mapfile.cpp
+
+#include "mapfile.h"
+#include <stdlib.h>
+#include <string.h>
+#include <string>
+#include <vector>
+#include <map>
+
+
+// class cmapfile_mem_entry (encapsulates mapfile_mem_entry_t)
+class cmapfile_mem_entry : public mapfile_mem_entry_t
+{
+public:
+	cmapfile_mem_entry(const cmapfile_mem_entry& e);
+	cmapfile_mem_entry(mapfile_mem_type_t t, uint32_t a, uint32_t s, const char* n);
+	~cmapfile_mem_entry();
+private:
+	void init(mapfile_mem_type_t t, uint32_t a, uint32_t s, const char* n);
+};
+
+// class cmapfile (encapsulates mapfile_t)
+class cmapfile : public mapfile_t
+{
+public:
+	cmapfile();
+	~cmapfile();
+	bool load(const char* fn);
+	cmapfile_mem_entry* get_mem_entry(const char* name);
```

Comment:
it is better to use stl-like names for functions like this - in this case "find"